### PR TITLE
docs: mongo databases

### PR DESCRIPTION
**Pull request by machi1990**

### CODE REVIEW

File: docs/databases/mongodb.md [hunk 1]
```diff
@@ -4,4 +4,99 @@ title: Using MongoDB in Graphback
...
+const modelDefs = loadSchemaSync(path.resolve('./User.graphql'), {
+loaders: [ new GraphQLFileLoader()]
+}); 
+
+// Use the dataProvider in buildGraphbackAPI to create resolvers
+const { resolvers, services, contextCreator } = buildGraphbackAPI(modelDefs, { dataProviderCreator });
+```
+
+The highlighted code does the following:
+ - Define connection URL to the MongoDB
```
Comment (craicoverflow) [inline docs/databases/mongodb.md:95]:
```suggestion
 - Define connection URL to the MongoDB.
```

File: docs/databases/mongodb.md [hunk 2]
```diff
@@ -4,4 +4,99 @@ title: Using MongoDB in Graphback
...
+ - We create a connection to our Mongo database.
+ - We select the `users` database
```
Comment (craicoverflow) [inline docs/databases/mongodb.md:97]:
```suggestion
 - We select the `users` database.
```

File: docs/databases/mongodb.md [hunk 3]
```diff
@@ -4,4 +4,99 @@ title: Using MongoDB in Graphback
...
+// load the schema
+const modelDefs = loadSchemaSync(path.resolve('./User.graphql'), {
+loaders: [ new GraphQLFileLoader()]
+}); 
+
+// Use the dataProvider in buildGraphbackAPI to create resolvers
+const { resolvers, services, contextCreator } = buildGraphbackAPI(modelDefs, { dataProviderCreator });
+```
+
+The highlighted code does the following:
```
Comment (craicoverflow) [inline docs/databases/mongodb.md:94]: The bullet points that follow switch from the perspective of "the code" is doing something to "we" do something.

I like when it's described from the code's perspective (like first point).

This is just my own preference, but whichever you choose, keep all bullet points from the same perspective.

File: docs/databases/mongodb.md [hunk 4]
```diff
@@ -4,4 +4,99 @@ title: Using MongoDB in Graphback
...
+
+// create the MongoDB Data provider creator function
+const dataProviderCreator = createMongoDbProvider(db);
+
+// highlight-end
+
+// load the schema
+const modelDefs = loadSchemaSync(path.resolve('./User.graphql'), {
+loaders: [ new GraphQLFileLoader()]
+});
```
[the remark this answers is not in the document]
Comment (machi1990) [reply docs/databases/mongodb.md:88]: You are right. Let's not clutter this part.

File: docs/databases/mongodb.md (hunk 3 above)
Comment (machi1990) [reply docs/databases/mongodb.md:94]: Good point/perspective.

File: docs/databases/mongodb.md [hunk 5]
```diff
@@ -4,4 +4,99 @@ title: Using MongoDB in Graphback
 sidebar_label: MongoDB 
 ---
 
-## TODO
\ No newline at end of file
+[MongoDB](https://www.mongodb.com/) is a general purpose, document-based, distributed database built for modern application developers.
```
[the remark this answers is not in the document]
Comment (machi1990) [reply docs/databases/mongodb.md:7]: I'd rather keep the link to npm since this will always point to the latest release which is not necessary the master.

**Comment by machi1990** [comment]
> Content is good, mostly left comments/suggestions.

Thanks for the thorough review @craicoverflow. Do you see this structure being replicated for PostgreSQL and SQLite (with a reco for dev / testing only) docs?

File: docs/databases/mongodb.md [hunk 6]
```diff
@@ -4,4 +4,101 @@ title: Using MongoDB in Graphback
...
+// get db instance
+const db: Db = mongoClient.db('users');
+
+// create the MongoDB Data provider creator function
+const dataProviderCreator = createMongoDbProvider(db);
+
+// highlight-end
+
+// the business model
+const userModel = ```
```
Comment (craicoverflow) [inline docs/databases/mongodb.md:82]:
```suggestion
const userModel = `
```

Should be a single "`" for a template literal

Comment (machi1990) [reply docs/databases/mongodb.md]: You can tell that I've been writing docs the whole day :-D

Comment (machi1990) [reply docs/databases/mongodb.md]: Good catch!

**Pull request merged**